src/main/java/hudson/plugins/git/util/DefaultBuildChooser.java

```diff
@@ -121,7 +127,41 @@ public DefaultBuildChooser() {
             }
         }
         
-        return revisions;
+        return sortByCriticalBranches(revisions, listener);
+    }
+    
+    private Collection<Revision> sortByCriticalBranches(Collection<Revision> revisions, TaskListener listener) {
+        Comparator<Revision> revisionComparator = new Comparator<Revision>() {
+			public int compare(Revision r1, Revision r2) {
+				return branchWithCritical(r2.getBranches()) - branchWithCritical(r1.getBranches());
+			}
+			
+			private int branchWithCritical(Collection<Branch> branches) {
+				for (Branch branch : branches) {
+					if (branch.getName().contains("CRITICAL")) {
```

Comment:
I agree, but it's a bigger change than you'd think. The correct way to implement this would be combine multiple BuildChoosers together, but currently the plugin only supports one BuildChooser and we are using the AncestryAgeBuildChooser for watershed commits.